src/tests/ftest/util/job_manager_utils.py

```diff
@@ -126,6 +121,38 @@ def assign_environment_default(self, env_vars):
         """
         pass
 
+    def get_subprocess_state(self, message=None):
+        """Display the state of the subprocess.
+
+        Args:
+            message (str, optional): additional text to include in output.
+                Defaults to None.
+
+        Returns:
+            list: a list of states for the process found. Multiple states should
+                indicate that the job manager and at least one remote process
+                are still running.
+
+        """
+        # Get/display the state of the job manager process
+        state = super(JobManager, self).get_subprocess_state(message)
+        if self._process is not None and self._hosts:
+            command = "/usr/bin/pgrep -a {}".format(self.job.command_regex)
+            self.log.debug(
+                "%s processes still running remotely%s:", self.command,
+                " {}".format(message) if message else "")
+            self.log.debug("Running (on %s): %s", self._hosts, command)
+            results = pcmd(self._hosts, command, True, 10, None)
+            # pgrep will return {1: <NodeSet>} if there are no remote processes
+            # running on any of the hosts.  If this value is not returned,
+            # indicate there are processes running by adding "R" states to the
+            # list.
+            if 1 not in results or len(results) > 1:
+                if not state:
+                    state = ["?"]
+                state.append("R")
```

Comment:
Is it feasible to replace R with REM or REMOTE to distinguish this from the actual process states (e.g., as can be found in the ps manual page) that are going to be reported for the job manager process (e.g., orterun with 'D' uninterrubtible sleep or 'Z' zombie state)?

Reply:
It was intentional to use a known state.  Since pgrep, as far as I can tell, has no way of returning the state of the remote processes I chose "R" since all we know is that the remote process is running.  We also know how many processes are running from the pgrep output, but for our current purposes we only need to know if at least one is running across all the remote hosts.  With the code logging the `ps` and `pgrep` output there is a sufficient amount of information to determine whether a `["R"]` return value is due to the existence of the local or remote process.